src/stores/Store.ts

```diff
@@ -280,15 +627,86 @@ export default class Store {
 	};
 
 	@action setSelectedSchemaType = (type: SchemaType) => {
+		localStorage.setItem('selectedSchemaType', type);
+		this.indicators = [];
 		this.selectedSchemaType = type;
+		this.prepareForSelectedSchemaType(type);
 		this.setIsSchemaApplied(false);
 	};
 
+	prepareForSelectedSchemaType = (type: SchemaType) => {
+		switch (type) {
+			case 'parsed-message':
+				{
+					const messageList = localStorage.getItem('parsedMessagesHistory')
+						? localStorage.getItem('parsedMessagesHistory')
+						: '';
+					localStorage.removeItem('parsedMessagesHistory');
+					this.parsedMessagesHistory = [];
+					if (messageList !== '') {
+						JSON.parse(messageList as string).forEach((element: string) => {
+							this.addParsedMessage(element as unknown as ParsedMessageItem);
+						});
+					}
+
+					this.setEditMessageMode(
+						localStorage.getItem('editParsedMessageMode') === 'true',
+					);
+
+					this.editedMessageIndex = localStorage.getItem('editedParsedMessageIndex')
+						? (localStorage.getItem('editedParsedMessageIndex') as unknown as number)
+						: -1;
+
+					this.setEditorProperties(
+						localStorage.getItem('selectedSessionId'),
+						localStorage.getItem('selectedDictionary'),
+						localStorage.getItem('selectedMessageType'),
+						this.editMessageMode
+							? (this.parsedMessagesHistory[this.editedMessageIndex]
+								.message as string)
+							: '{}',
+					);
+				}
+				break;
+			case 'act':
+				{
+					const actMessageList = localStorage.getItem('actMessagesHistory')
```

Comment:
there are too much of localStorage.getItem,  localStorage.setItem calls

i think localstorage should only be used for persisting state, it shouldn't be source of state.
Meaning you should only be getting values from localStorage on app start to initialize state. that's it.
After that you should only keep mobx state and localStorage synced. And much easier way to do it is to use autorun or reaction.
for e.g `reaction(() => this.data, data => localStorage.set(key, data))`
this will run every time you change this data